source/backend/api/Services/FormDocumentService.cs

```diff
@@ -0,0 +1,136 @@
+using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using MapsterMapper;
+using Microsoft.Extensions.Logging;
+using Pims.Api.Models;
+using Pims.Api.Models.Concepts;
+using Pims.Dal.Entities;
+using Pims.Dal.Helpers.Extensions;
+using Pims.Dal.Repositories;
+using Pims.Dal.Security;
+
+namespace Pims.Api.Services
+{
+    public class FormDocumentService : BaseService, IFormDocumentService
+    {
+        private readonly ILogger _logger;
+        private readonly IMapper _mapper;
+        private readonly IFormTypeRepository _formTypeRepository;
+        private readonly IDocumentService _documentService;
+        private readonly IDocumentRepository _documentRepository;
+        private readonly IAcquisitionFileFormRepository _acquisitionFileFormRepository;
+
+        public FormDocumentService(
+            ClaimsPrincipal user,
+            ILogger<FormDocumentService> logger,
+            IMapper mapper,
+            IFormTypeRepository formTypeRepository,
+            IDocumentService documentService,
+            IDocumentRepository documentRepository,
+            IAcquisitionFileFormRepository acquisitionFileFormRepository)
+            : base(user, logger)
+        {
+            _logger = logger;
+            _mapper = mapper;
+            _formTypeRepository = formTypeRepository;
+            _documentService = documentService;
+            _documentRepository = documentRepository;
+            _acquisitionFileFormRepository = acquisitionFileFormRepository;
+        }
+
+        public IList<PimsFormType> GetAllFormDocumentTypes()
+        {
+            _logger.LogInformation("Getting all form document types");
+            this.User.ThrowIfNotAuthorized(Permissions.FormView);
+
+            var formTemplates = _formTypeRepository.GetAllFormTypes();
+            return formTemplates;
+        }
+
+        public IList<PimsFormType> GetFormDocumentTypes(string formTypeCode)
+        {
+            this.Logger.LogInformation("Getting document types for code {formTypeCode}", formTypeCode);
+            this.User.ThrowIfNotAuthorized(Permissions.FormView);
+
+            return new List<PimsFormType>() { _formTypeRepository.GetByFormTypeCode(formTypeCode) };
+        }
+
+        public async Task<DocumentUploadRelationshipResponse> UploadFormDocumentTemplateAsync(string formTypeCode, DocumentUploadRequest uploadRequest)
+        {
+            this.Logger.LogInformation("Uploading template for document form type");
+            this.User.ThrowIfNotAuthorized(Permissions.DocumentAdmin);
+
+            var currentFormType = _formTypeRepository.GetByFormTypeCode(formTypeCode);
+            if (currentFormType.DocumentId != null)
+            {
+                var result = await DeleteFormDocumentTemplateAsync(currentFormType);
+                if (result.Status != ExternalResultStatus.Success)
```

Comment:
Is there an advantage to using ExternalResult here instead of a boolean? (asking because I used a boolean for a delete on my story but would standardize if there was a benefit).

Reply:
It is more flexible. The caller would have more knowledge about why the call failed.

Reply:
oh wait I'm being thick, this is a call to Mayan not to the repo, sorry.